torch/jit/_recursive.py

```diff
@@ -127,6 +127,12 @@ def infer_type(name, item):
 
         added_names.add(name)
 
+    for idx, fn in nn_module._forward_hooks.items():
```

Comment:
Here we should be compiling the hooks to the same compilation unit as the module being compiled, then get the qualified names, then store those on the module. This would let us share hooks if they are re-used between modules which more closely matches how they are used in Python. It looks like all the other plumbing to make this work is already there in the rest of this PR